RevenueCat/Plugins/iOS/PurchasesUnityHelper.m

```diff
@@ -58,17 +58,23 @@ - (void)setupPurchases:(NSString *)apiKey
              appUserID:(nullable NSString *)appUserID
             gameObject:(NSString *)gameObject
           observerMode:(BOOL)observerMode
- userDefaultsSuiteName:(nullable NSString *)userDefaultsSuiteName {
-    [[RCPurchases sharedPurchases] setDelegate:nil];
+ userDefaultsSuiteName:(nullable NSString *)userDefaultsSuiteName
+     dangerousSettings:(NSDictionary *)dangerousSettingsAsDictionary {
+    if (RCPurchases.isConfigured) {
+        [[RCPurchases sharedPurchases] setDelegate:nil];
+    }
```

Comment:
This was a tricky one to realize. The newer version of iOS throws an exception if `sharedPurchases` is accessed before configuring.